improver/wxcode/weather_symbols.py

```diff
@@ -452,35 +373,26 @@ def _constraint_string(diagnostic, threshold_name, threshold_val):
                     Name of threshold coordinate on input cubes
                 threshold_val (float):
                     Value of threshold coordinate required
-            Returns: (str)
+            Returns: iris.Constraint
             """
             if abs(threshold_val) < WeatherSymbols().float_abs_tolerance:
-                cell_constraint_str = (
-                    " -{float_abs_tol} < cell < "
-                    "{float_abs_tol}".format(
-                        float_abs_tol=WeatherSymbols().float_abs_tolerance
-                    )
-                )
+                float_abs_tol = WeatherSymbols().float_abs_tolerance
+                cell_constraint = lambda cell: -float_abs_tol < cell < float_abs_tol
             else:
-                cell_constraint_str = (
-                    "{threshold_val} * {float_min} < cell < "
-                    "{threshold_val} * {float_max}".format(
-                        threshold_val=threshold_val,
-                        float_min=(1.0 - WeatherSymbols().float_tolerance),
-                        float_max=(1.0 + WeatherSymbols().float_tolerance),
-                    )
+                float_min = 1.0 - WeatherSymbols().float_tolerance
+                float_max = 1.0 + WeatherSymbols().float_tolerance
+                cell_constraint = (
+                    lambda cell: threshold_val * float_min
+                    < cell
+                    < threshold_val * float_max
```

Comment:
Can we make use of `np.isclose` with relative tolerances here?  Might be tidier.

Reply:
Fixed.